packages/core/src/Cardano/types/CardanoNodeErrors.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  AcquirePointNotOnChainError,
+  AcquirePointTooOldError,
+  EraMismatchError,
+  IntersectionNotFoundError,
+  QueryUnavailableInCurrentEraError,
+  ServerNotReady,
+  TipIsOriginError,
+  UnknownResultError,
+  WebSocketClosed
+} from '@cardano-ogmios/client';
+import { CustomError } from 'ts-custom-error';
+
+export class UnknownCardanoNodeError extends CustomError {
+  constructor(public innerError: unknown) {
+    super('Unknown CardanoNode error. See "innerError".');
+  }
+}
+
+export class CardanoNodeNotInitializedError extends CustomError {
+  constructor(methodName: string) {
+    super(`${methodName} cannot be called until CardanoNode is initialized`);
+  }
+}
+
+export const CardanoNodeErrors = {
+  AcquirePointNotOnChainError,
+  AcquirePointTooOldError,
+  EraMismatchError,
+  IntersectionNotFoundError,
+  QueryUnavailableInCurrentEraError,
+  ServerNotReady,
+  TipIsOriginError,
+  UnknownResultError,
+  WebSocketClosed
+};
+
+type CardanoNodeErrorName = keyof typeof CardanoNodeErrors;
+type CardanoNodeErrorClass = typeof CardanoNodeErrors[CardanoNodeErrorName];
+export type CardanoNodeError = InstanceType<CardanoNodeErrorClass> | UnknownCardanoNodeError;
```

Comment:
Any reason not to do this to have a single type that we can use in doc comments and users can cast to in `catch`?

```suggestion
export type CardanoNodeError = InstanceType<CardanoNodeErrorClass> | UnknownCardanoNodeError | CardanoNodeNotInitializedError;
```